multi_import/object_cache.py

```diff
@@ -32,37 +33,67 @@ def __init__(self, lookup_fields):
     def _default_factory(self):
         return defaultdict(set)
 
-    def _add_lookup_value(self, field, value, instance):
+    def _add_lookup_value(self, field, value, cached_object):
         if not value:
             return
 
         lookup_dict = self[field]
         key = unicode(value)
 
-        lookup_dict[key].add(
-            CachedObject(instance)
-        )
+        lookup_dict[key].add(cached_object)
+
+    def _check_instance_set(self, instance_set):
+        if len(instance_set) > 1:
+            raise self.multiple_objects_error
+        return None
 
     def _get_lookup_value(self, field, value):
-        if field not in self or value is None:
-            return None
+        if isinstance(field, six.string_types):
+            field = (field,)
+            value = (value,)
 
-        lookup_dict = self[field]
-        key = unicode(value)
+        zipped_values = list(zip(field, value))
+
+        for f, v in zipped_values:
+            if f not in self or v is None:
+                return None
+
+        instance_sets = [
+            self[f][unicode(v)]
+            for f, v in zipped_values
+        ]
+
+        for s in instance_sets:
+            if not s:
+                return None
+
+        instance_set = set.intersection(*instance_sets)
 
-        instance_set = lookup_dict[key]
         if not instance_set:
             return None
 
-        if len(instance_set) > 1:
-            raise self.multiple_objects_error
+        result = self._check_instance_set(instance_set)
+        if result:
+            return result
 
         return next(iter(instance_set)).obj
 
     def cache_instance(self, instance):
-        for field in self.lookup_fields:
+        fields_to_cache = set(
+            item for sublist in
+            (
+                (fields,) if isinstance(fields, six.string_types) else fields
+                for fields in self.lookup_fields
+            )
+            for item in sublist
```

Comment:
This flattens a list of lists.

Example:

```python
a = [
    [1, 2, 3],
    [4, 5, 6]
]

b = [item for sublist in a for item in sublist]

print b # [1, 2, 3, 4, 5, 6]
```

See: http://stackoverflow.com/a/952952